src/Schema/BlueprintLaravelCompatibility.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MongoDB\Laravel\Schema;
+
+use Closure;
+use Illuminate\Database\Connection;
+use Illuminate\Database\Schema\Blueprint as BaseBlueprint;
+
+use function property_exists;
+
+/**
+ * The $connection property and constructor param were added in Laravel 12
+ * We keep the untyped $connection property for older version of Laravel to maintain compatibility
+ * and not break projects that would extend the MongoDB Blueprint class.
+ *
+ * @see https://github.com/laravel/framework/commit/f29df4740d724f1c36385c9989569e3feb9422df#diff-68f714a9f1b751481b993414d3f1300ad55bcef12084ec0eb8f47f350033c24bR107
+ */
+if (! property_exists(BaseBlueprint::class, 'connection')) {
+    /** @internal For compatibility with Laravel 10 and 11 */
+    trait BlueprintLaravelCompatibility
+    {
+        /**
+         * The MongoDB connection object for this blueprint.
+         *
+         * @var Connection
+         */
+        protected $connection;
```

Comment:
Does this also need to be declared in the Laravel 12+ trait?

The earlier version of this PR also added a type hint on the `$collection` property (https://github.com/mongodb/laravel-mongodb/pull/3283#discussion_r1971818758). I'm not sure if you intended to include that here.

Reply:
That's not necessary as the parent class declares it. And we don't use it in the trait.